kivy/uix/modalview.py

```diff
@@ -186,70 +189,49 @@ def __init__(self, **kwargs):
         self._parent = None
         super(ModalView, self).__init__(**kwargs)
 
-    def _search_window(self):
-        # get window to attach to
-        window = None
-        if self.attach_to is not None:
-            window = self.attach_to.get_parent_window()
-            if not window:
-                window = self.attach_to.get_root_window()
-        if not window:
-            from kivy.core.window import Window
-            window = Window
-        return window
-
-    def open(self, *largs, **kwargs):
-        '''Show the view window from the :attr:`attach_to` widget. If set, it
-        will attach to the nearest window. If the widget is not attached to any
-        window, the view will attach to the global
-        :class:`~kivy.core.window.Window`.
+    def open(self, *_args, **kwargs):
+        """ attach this view to the main window.
 
         When the view is opened, it will be faded in with an animation. If you
         don't want the animation, use::
 
             view.open(animation=False)
 
-        '''
-        if self._window is not None:
-            Logger.warning('ModalView: you can only open once.')
-            return
-        # search window
-        self._window = self._search_window()
-        if not self._window:
-            Logger.warning('ModalView: cannot open view, no window found.')
-            return
+        """
+        self._window = Window
```

Comment:
```suggestion
        """
        if self._is_open:
            return
        self._window = Window
```
Don't we still need to check if it's already open?

Reply:
good catch